src/client/core/extensions/utils.js

```diff
@@ -0,0 +1,65 @@
+import hammerhead from '../deps/hammerhead';
+
+
+export const CONTAINS_OWN_TEXT_METHOD_NAME = 'containsExcludeChildren';
+export const ATTR_REGEXP_METHOD_NAME       = 'attrRegExp';
+
+const TEXT_ELEMENT_TAGS_REGEXP      = /^i$|^b$|^big$|^small$|^em$|^strong$|^dfn$|^code$|^samp$|^kbd$|^var$|^cite$|^abbr$|^acronym$|^sub$|^sup$|span$|^bdo$|^address$|^div$|^a$|^object$|^p$|^h\d$|^pre$|^q$|^ins$|^del$|^dt$|^dd$|^li$|^label$|^option$|^textarea$|^fieldset$|^legend$|^button$|^caption$|^td$|^th$|^title$/;
+const REGEXP_START_SUBSTR           = ':/';
+const REGEXP_END_SUBSTR             = '/';
+
+function getOwnTextForSelector (element) {
+    if (!element || !TEXT_ELEMENT_TAGS_REGEXP.test(element.tagName.toLowerCase()))
+        return '';
+
+    var text = '';
+
+    for (var i = 0; i < element.childNodes.length; i++) {
+        if (element.childNodes[i].nodeType === 3 && element.childNodes[i].data)
+            text += element.childNodes[i].data;
+    }
+
+    if (text) {
+        //\xC0-\xFF - latin letters
+        text = hammerhead.utils.trim(text.replace(/[^a-z0-9а-я \xC0-\xFF]+/gi, ''));
+        text = text.replace(/\s+/g, ' ');
+        if (/\S/.test(text))
+            return text;
+    }
+
+    return '';
+}
+
+export function containsExcludeChildrenPredicate (el, selector) {
+    if (!selector)
+        return false;
+
+    var textExcludeChildren = getOwnTextForSelector(el);
+
+    return !!(textExcludeChildren && textExcludeChildren.indexOf(selector) >= 0);
```

Comment:
No need to `!!()` because `textExcludeChildren.indexOf(selector) >= 0` returns `boolean`